### PR TITLE
(SOFTWARE3285) Adding the steps to build RPM out of GitHub

### PR DESCRIPTION
**Pull request by efajardo**

### CODE REVIEW

File: docs/software/rpm-development-guide.md [hunk 1]
```diff
@@ -247,6 +247,19 @@ where:
 In addition, if the repository contains a file called `rpm/<PROJECT>.spec`, it will be used as the spec file for the build
 (unless overridden in the `osg` directory).
 
+##### Typycal workflow when building out of Git and GitHub repos
```
Comment (brianhlin) [inline docs/software/rpm-development-guide.md]:
```suggestion
##### Typical workflow when building out of GitHub repos
```

File: docs/software/rpm-development-guide.md [hunk 2]
```diff
@@ -247,6 +247,19 @@ where:
...
+
+1. Fork the repository that you would like to build for
+1. In your fork, tag the commit yoy would like to try building
+1. In the upstream/osg.github.source, change the repo to point at your fork and tag
```
Comment (brianhlin) [inline docs/software/rpm-development-guide.md]:
```suggestion
1. In the `upstream/osg.github.source`, change the repo to point at your fork and tag
```

File: docs/software/rpm-development-guide.md [hunk 3]
```diff
@@ -247,6 +247,19 @@ where:
 In addition, if the repository contains a file called `rpm/<PROJECT>.spec`, it will be used as the spec file for the build
 (unless overridden in the `osg` directory).
 
+##### Typycal workflow when building out of Git and GitHub repos
+
+1. Fork the repository that you would like to build for
+1. In your fork, tag the commit yoy would like to try building
```
Comment (brianhlin) [inline docs/software/rpm-development-guide.md]:
```suggestion
1. In your fork, tag the commit you would like to try building
```

File: docs/software/rpm-development-guide.md [hunk 4]
```diff
@@ -247,6 +247,19 @@ where:
...
+1. In the upstream/osg.github.source, change the repo to point at your fork and tag
+1. Attempt a scratch build
+1. If the build fails, remove the tag and repeat steps 2-4
+1. Submit a PR to merge changes upstream
+1. Tag the final version on the upstream fork
+1. Build the version that will go through the normal software cycle
+
+!!! note
+    Packaging only changes can be handled with a `v<version>-<release>` tag
```
Comment (brianhlin) [inline docs/software/rpm-development-guide.md]:
```suggestion
    Packaging-only changes should be tagged `v<version>-<release>`, i.e. `v3.4.23-2`
```

File: docs/software/rpm-development-guide.md [hunk 5]
```diff
@@ -247,6 +247,19 @@ where:
 In addition, if the repository contains a file called `rpm/<PROJECT>.spec`, it will be used as the spec file for the build
 (unless overridden in the `osg` directory).
 
+##### Typycal workflow when building out of Git and GitHub repos
+
+1. Fork the repository that you would like to build for
```
Comment (brianhlin) [inline docs/software/rpm-development-guide.md]: I think we're missing a step after this one that says something about making and committing changes to the repository. Once you add that, the step that says "repeat steps 2-5" should say "repeat steps 2-5".

**Comment by efajardo** [comment]
Thanks for the review @brianhlin  I addressed your comments.

**Comment by brianhlin** [comment]
CI failures unrelated, merging

**Pull request merged**